In [1]:
import os
import preprocessing
import vggish_processing
import mfcc_stat
from tqdm import tqdm
import pandas as pd
import numpy as np
import librosa
import librosa.display

In [2]:
root = "/Extracted_data/"
dirs = os.listdir(root)
data = preprocessing.Get_Data(dirs,root)
cough_heavy = data.get_cough_heavy()

100%|████████████████████████████████████████████████████████████████████████████████| 225/225 [01:06<00:00,  3.36it/s]


In [3]:
cough_heavy.head()

label                                               data  \
0    0.0  [1.141661e-05, 3.862317e-06, -3.9919647e-05, -...   
1    1.0  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
2    1.0  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
3    0.0  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
4    0.0  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   

                                            filename  
0  C:/Users/deepd/Documents/Covid/Extracted_trial...  
1  C:/Users/deepd/Documents/Covid/Extracted_trial...  
2  C:/Users/deepd/Documents/Covid/Extracted_trial...  
3  C:/Users/deepd/Documents/Covid/Extracted_trial...  
4  C:/Users/deepd/Documents/Covid/Extracted_trial...

## Extracting MFCC feature

In [4]:
Extracted_feature = preprocessing.Extract_Data(cough_heavy)

In [5]:
# extracting three variants of MFFCs
mfcc = Extracted_feature.get_mfcc()
mfccdelta = Extracted_feature.get_deltamfcc()
mfcc2delta = Extracted_feature.get_delta2mfcc()

## Computing different statistics from MFCC

In [6]:
# computing a total of 11 statistics from the first 13 components of MFCC
mfcc_feature = mfcc_stat.mfcc_stat(mfcc)
mfcc_mean = mfcc_feature.mean()
df_mfcc_mean = pd.DataFrame(mfcc_mean,columns=['mfcc_mean_feature']*13)
df_mfcc_feature = df_mfcc_mean

mfcc_std = mfcc_feature.std()
df_mfcc_std = pd.DataFrame(mfcc_std,columns=['mfcc_std_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_std)

mfcc_iqr = mfcc_feature.interquartilerange()
df_mfcc_iqr = pd.DataFrame(mfcc_iqr,columns=['mfcc_iqr_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_iqr)

mfcc_kurto = mfcc_feature.kurtosis()
df_mfcc_kurto = pd.DataFrame(mfcc_kurto,columns=['mfcc_kurto_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_kurto)

mfcc_max = mfcc_feature.maximum()
df_mfcc_max = pd.DataFrame(mfcc_max,columns=['mfcc_max_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_max)

mfcc_min = mfcc_feature.minimum()
df_mfcc_min = pd.DataFrame(mfcc_min,columns=['mfcc_min_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_min)

mfcc_median = mfcc_feature.median()
df_mfcc_median = pd.DataFrame(mfcc_median,columns=['mfcc_median_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_median)

mfcc_qtl1 = mfcc_feature.quartile_1st()
df_mfcc_qtl1 = pd.DataFrame(mfcc_qtl1,columns=['mfcc_quartile1_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_qtl1)

mfcc_qtl3 = mfcc_feature.quartile_3rd()
df_mfcc_qtl3 = pd.DataFrame(mfcc_qtl3,columns=['mfcc_quartile3_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_qtl3)

mfcc_skew = mfcc_feature.skewness()
df_mfcc_skew = pd.DataFrame(mfcc_skew,columns=['mfcc_skewness_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_skew)

mfcc_rms = mfcc_feature.get_rms()
df_mfcc_rms = pd.DataFrame(mfcc_rms,columns=['mfcc_rms_feature']*13)
df_mfcc_feature = df_mfcc_feature.join(df_mfcc_rms)


In [7]:
df_mfcc_feature.head().T

0           1           2           3           4
mfcc_mean_feature -765.857361 -673.794434 -718.185974 -771.010437 -652.407043
mfcc_mean_feature   56.635921   39.070114   66.435989   65.506836   99.792557
mfcc_mean_feature   -7.718951  -55.959335   -7.685933   10.993238  -42.095676
mfcc_mean_feature   -5.349398   -3.420140   21.239208   14.352894   30.789314
mfcc_mean_feature   -2.479132  -32.085152  -40.569927  -20.039225  -26.789732
...                       ...         ...         ...         ...         ...
mfcc_rms_feature    13.979434   39.289099   12.166927   11.493811    8.813728
mfcc_rms_feature     6.377627   11.366460   30.442542   13.170262   16.047941
mfcc_rms_feature    14.007841   10.699667   11.237177    7.029454   19.779757
mfcc_rms_feature     6.766296   21.027959    9.660470   12.305389   10.055729
mfcc_rms_feature     8.104084   16.366894   14.450569    7.089626    8.191580

[143 rows x 5 columns]

In [8]:
# computing a total of 11 statistics from the first 13 components of Delta MFCC
mfccdelta_feature = mfcc_stat.mfcc_stat(mfccdelta)
mfccdelta_mean = mfccdelta_feature.mean()
df_mfccdelta_mean = pd.DataFrame(mfccdelta_mean,columns=['mfccdelta_mean_feature']*13)
df_mfccdelta_feature = df_mfccdelta_mean

mfccdelta_std = mfccdelta_feature.std()
df_mfccdelta_std = pd.DataFrame(mfccdelta_std,columns=['mfccdelta_std_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_std)

mfccdelta_iqr = mfccdelta_feature.interquartilerange()
df_mfccdelta_iqr = pd.DataFrame(mfccdelta_iqr,columns=['mfccdelta_iqr_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_iqr)

mfccdelta_kurto = mfccdelta_feature.kurtosis()
df_mfccdelta_kurto = pd.DataFrame(mfccdelta_kurto,columns=['mfccdelta_kurto_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_kurto)

mfccdelta_max = mfccdelta_feature.maximum()
df_mfccdelta_max = pd.DataFrame(mfccdelta_max,columns=['mfccdelta_max_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_max)

mfccdelta_min = mfccdelta_feature.minimum()
df_mfccdelta_min = pd.DataFrame(mfccdelta_min,columns=['mfccdelta_min_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_min)

mfccdelta_median = mfccdelta_feature.median()
df_mfccdelta_median = pd.DataFrame(mfccdelta_median,columns=['mfccdelta_median_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_median)

mfccdelta_qtl1 = mfccdelta_feature.quartile_1st()
df_mfccdelta_qtl1 = pd.DataFrame(mfccdelta_qtl1,columns=['mfccdelta_quartile1_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_qtl1)

mfccdelta_qtl3 = mfccdelta_feature.quartile_3rd()
df_mfccdelta_qtl3 = pd.DataFrame(mfccdelta_qtl3,columns=['mfccdelta_quartile3_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_qtl3)

mfccdelta_skew = mfccdelta_feature.skewness()
df_mfccdelta_skew = pd.DataFrame(mfccdelta_skew,columns=['mfccdelta_skewness_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_skew)

mfccdelta_rms = mfccdelta_feature.get_rms()
df_mfccdelta_rms = pd.DataFrame(mfccdelta_rms,columns=['mfccdelta_rms_feature']*13)
df_mfccdelta_feature = df_mfccdelta_feature.join(df_mfccdelta_rms)


In [9]:
df_mfccdelta_feature.head().T

0         1         2         3         4
mfccdelta_mean_feature -0.146090  0.310150 -0.068103  0.181848  0.726761
mfccdelta_mean_feature -0.151706  0.451770 -0.305376  0.121666  0.389425
mfccdelta_mean_feature -0.040220 -0.036425 -0.092710  0.044122 -0.087469
mfccdelta_mean_feature  0.023623 -0.346961 -0.001072 -0.061309  0.030319
mfccdelta_mean_feature  0.006809 -0.082435 -0.134195 -0.130874 -0.122579
...                          ...       ...       ...       ...       ...
mfccdelta_rms_feature   1.741584  4.475665  1.729634  1.713301  1.479621
mfccdelta_rms_feature   1.178085  2.286302  3.076882  1.841228  2.043853
mfccdelta_rms_feature   2.442429  2.070615  1.900294  1.315970  2.131574
mfccdelta_rms_feature   1.018237  2.701472  1.435105  1.530184  1.896406
mfccdelta_rms_feature   1.170920  2.606992  2.139350  1.253541  1.485487

[143 rows x 5 columns]

In [10]:
# computing a total of 11 statistics from the first 13 components of Delta2 MFCC
mfccdelta2_feature = mfcc_stat.mfcc_stat(mfcc2delta)
mfccdelta2_mean = mfccdelta2_feature.mean()
df_mfccdelta2_mean = pd.DataFrame(mfccdelta2_mean,columns=['mfccdelta2_mean_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_mean

mfccdelta2_std = mfccdelta2_feature.std()
df_mfccdelta2_std = pd.DataFrame(mfccdelta2_std,columns=['mfccdelta2_std_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_std)

mfccdelta2_iqr = mfccdelta2_feature.interquartilerange()
df_mfccdelta2_iqr = pd.DataFrame(mfccdelta2_iqr,columns=['mfccdelta2_iqr_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_iqr)

mfccdelta2_kurto = mfccdelta2_feature.kurtosis()
df_mfccdelta2_kurto = pd.DataFrame(mfccdelta2_kurto,columns=['mfccdelta2_kurto_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_kurto)

mfccdelta2_max = mfccdelta2_feature.maximum()
df_mfccdelta2_max = pd.DataFrame(mfccdelta2_max,columns=['mfccdelta2_max_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_max)

mfccdelta2_min = mfccdelta2_feature.minimum()
df_mfccdelta2_min = pd.DataFrame(mfccdelta2_min,columns=['mfccdelta2_min_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_min)

mfccdelta2_median = mfccdelta2_feature.median()
df_mfccdelta2_median = pd.DataFrame(mfccdelta2_median,columns=['mfccdelta2_median_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_median)

mfccdelta2_qtl1 = mfccdelta2_feature.quartile_1st()
df_mfccdelta2_qtl1 = pd.DataFrame(mfccdelta2_qtl1,columns=['mfccdelta2_quartile1_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_qtl1)

mfccdelta2_qtl3 = mfccdelta2_feature.quartile_3rd()
df_mfccdelta2_qtl3 = pd.DataFrame(mfccdelta2_qtl3,columns=['mfccdelta2_quartile3_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_qtl3)

mfccdelta2_skew = mfccdelta2_feature.skewness()
df_mfccdelta2_skew = pd.DataFrame(mfccdelta2_skew,columns=['mfccdelta2_skewness_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_skew)

mfccdelta2_rms = mfccdelta2_feature.get_rms()
df_mfccdelta2_rms = pd.DataFrame(mfccdelta2_rms,columns=['mfccdelta2_rms_feature']*13)
df_mfccdelta2_feature = df_mfccdelta2_feature.join(df_mfccdelta2_rms)


In [11]:
df_mfccdelta2_feature.head().T

0         1         2         3         4
mfccdelta2_mean_feature -0.012415  0.082967 -0.258965  0.064679 -0.293373
mfccdelta2_mean_feature -0.011710  0.188356 -0.302228  0.041547 -0.024611
mfccdelta2_mean_feature  0.004304  0.038664 -0.068151  0.024756  0.109241
mfccdelta2_mean_feature  0.016600 -0.008935  0.017337 -0.003380 -0.022185
mfccdelta2_mean_feature  0.011651  0.125034  0.016279 -0.024954  0.063565
...                           ...       ...       ...       ...       ...
mfccdelta2_rms_feature   1.005784  2.104446  1.105345  1.083020  1.162813
mfccdelta2_rms_feature   0.783915  1.342029  1.849997  1.220935  1.075796
mfccdelta2_rms_feature   1.303082  1.365444  1.126090  0.840846  1.301782
mfccdelta2_rms_feature   0.693466  1.800500  0.961305  0.945776  1.127132
mfccdelta2_rms_feature   0.675729  1.422219  1.161680  0.703270  1.111508

[143 rows x 5 columns]

In [12]:
# joining the MFCC features into one dataframe
mfcc_feature = df_mfcc_feature.join(df_mfccdelta_feature)
mfcc_feature = mfcc_feature.join(df_mfccdelta2_feature)

In [13]:
mfcc_feature.head().T

0           1           2           3  \
mfcc_mean_feature      -765.857361 -673.794434 -718.185974 -771.010437   
mfcc_mean_feature        56.635921   39.070114   66.435989   65.506836   
mfcc_mean_feature        -7.718951  -55.959335   -7.685933   10.993238   
mfcc_mean_feature        -5.349398   -3.420140   21.239208   14.352894   
mfcc_mean_feature        -2.479132  -32.085152  -40.569927  -20.039225   
...                            ...         ...         ...         ...   
mfccdelta2_rms_feature    1.005784    2.104446    1.105345    1.083020   
mfccdelta2_rms_feature    0.783915    1.342029    1.849997    1.220935   
mfccdelta2_rms_feature    1.303082    1.365444    1.126090    0.840846   
mfccdelta2_rms_feature    0.693466    1.800500    0.961305    0.945776   
mfccdelta2_rms_feature    0.675729    1.422219    1.161680    0.703270   

                                 4  
mfcc_mean_feature      -652.407043  
mfcc_mean_feature        99.792557  
mfcc_mean_feature       -42.095676  
mfcc_mean_feature        30.789314  
mfcc_mean_feature       -26.789732  
...                            ...  
mfccdelta2_rms_feature    1.162813  
mfccdelta2_rms_feature    1.075796  
mfccdelta2_rms_feature    1.301782  
mfccdelta2_rms_feature    1.127132  
mfccdelta2_rms_feature    1.111508  

[429 rows x 5 columns]

In [14]:
mfcc_feature.shape

(225, 429)

In [15]:
mfcc_feature.to_csv('mfcc_features.csv',index=False)

## Extracting VGGISH features and statistics

In [16]:
vggish = [vggish_processing.vggish_embedding(file) for file in tqdm(cough_heavy['filename'])]

  0%|                                                                                          | 0/225 [00:00<?, ?it/s]

Instructions for updating:
Please use `layer.__call__` method instead.
Instructions for updating:
Use keras.layers.Flatten instead.
INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  0%|▎                                                                                 | 1/225 [00:04<15:00,  4.02s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  1%|▋                                                                                 | 2/225 [00:06<13:02,  3.51s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  1%|█                                                                                 | 3/225 [00:08<11:25,  3.09s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  2%|█▍                                                                                | 4/225 [00:10<10:23,  2.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  2%|█▊                                                                                | 5/225 [00:12<09:45,  2.66s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  3%|██▏                                                                               | 6/225 [00:15<09:16,  2.54s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  3%|██▌                                                                               | 7/225 [00:17<08:54,  2.45s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  4%|██▉                                                                               | 8/225 [00:19<08:39,  2.39s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  4%|███▎                                                                              | 9/225 [00:21<08:19,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  4%|███▌                                                                             | 10/225 [00:23<08:03,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  5%|███▉                                                                             | 11/225 [00:26<08:11,  2.30s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  5%|████▎                                                                            | 12/225 [00:28<07:57,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  6%|████▋                                                                            | 13/225 [00:30<08:06,  2.30s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  6%|█████                                                                            | 14/225 [00:32<07:42,  2.19s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  7%|█████▍                                                                           | 15/225 [00:35<07:42,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  7%|█████▊                                                                           | 16/225 [00:37<07:40,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  8%|██████                                                                           | 17/225 [00:39<07:36,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  8%|██████▍                                                                          | 18/225 [00:41<07:51,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  8%|██████▊                                                                          | 19/225 [00:44<07:48,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  9%|███████▏                                                                         | 20/225 [00:46<07:41,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


  9%|███████▌                                                                         | 21/225 [00:48<07:50,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 10%|███████▉                                                                         | 22/225 [00:51<07:44,  2.29s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 10%|████████▎                                                                        | 23/225 [00:53<07:39,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 11%|████████▋                                                                        | 24/225 [00:55<07:26,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 11%|█████████                                                                        | 25/225 [00:57<07:22,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 12%|█████████▎                                                                       | 26/225 [00:59<07:16,  2.19s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 12%|█████████▋                                                                       | 27/225 [01:01<07:18,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 12%|██████████                                                                       | 28/225 [01:04<07:12,  2.19s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 13%|██████████▍                                                                      | 29/225 [01:06<07:10,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 13%|██████████▊                                                                      | 30/225 [01:08<07:08,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 14%|███████████▏                                                                     | 31/225 [01:10<07:12,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 14%|███████████▌                                                                     | 32/225 [01:13<07:11,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 15%|███████████▉                                                                     | 33/225 [01:15<07:16,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 15%|████████████▏                                                                    | 34/225 [01:17<07:09,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 16%|████████████▌                                                                    | 35/225 [01:20<07:13,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 16%|████████████▉                                                                    | 36/225 [01:22<07:04,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 16%|█████████████▎                                                                   | 37/225 [01:24<07:10,  2.29s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 17%|█████████████▋                                                                   | 38/225 [01:26<06:59,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 17%|██████████████                                                                   | 39/225 [01:28<06:57,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 18%|██████████████▍                                                                  | 40/225 [01:31<07:07,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 18%|██████████████▊                                                                  | 41/225 [01:33<06:48,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 19%|███████████████                                                                  | 42/225 [01:35<06:42,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 19%|███████████████▍                                                                 | 43/225 [01:37<06:39,  2.19s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 20%|███████████████▊                                                                 | 44/225 [01:39<06:33,  2.17s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 20%|████████████████▏                                                                | 45/225 [01:41<06:16,  2.09s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 20%|████████████████▌                                                                | 46/225 [01:44<06:24,  2.15s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 21%|████████████████▉                                                                | 47/225 [01:46<06:40,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 21%|█████████████████▎                                                               | 48/225 [01:48<06:41,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 22%|█████████████████▋                                                               | 49/225 [01:51<06:45,  2.30s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 22%|██████████████████                                                               | 50/225 [01:53<06:34,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 23%|██████████████████▎                                                              | 51/225 [01:55<06:34,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 23%|██████████████████▋                                                              | 52/225 [01:57<06:24,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 24%|███████████████████                                                              | 53/225 [01:59<06:14,  2.18s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 24%|███████████████████▍                                                             | 54/225 [02:02<06:28,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 24%|███████████████████▊                                                             | 55/225 [02:04<06:33,  2.32s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 25%|████████████████████▏                                                            | 56/225 [02:07<06:29,  2.30s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 25%|████████████████████▌                                                            | 57/225 [02:09<06:27,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 26%|████████████████████▉                                                            | 58/225 [02:11<06:18,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 26%|█████████████████████▏                                                           | 59/225 [02:13<06:13,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 27%|█████████████████████▌                                                           | 60/225 [02:15<06:11,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 27%|█████████████████████▉                                                           | 61/225 [02:17<05:56,  2.17s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 28%|██████████████████████▎                                                          | 62/225 [02:20<05:52,  2.16s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 28%|██████████████████████▋                                                          | 63/225 [02:22<06:02,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 28%|███████████████████████                                                          | 64/225 [02:24<06:03,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 29%|███████████████████████▍                                                         | 65/225 [02:27<06:03,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 29%|███████████████████████▊                                                         | 66/225 [02:29<05:54,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 30%|████████████████████████                                                         | 67/225 [02:31<05:56,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 30%|████████████████████████▍                                                        | 68/225 [02:33<05:52,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 31%|████████████████████████▊                                                        | 69/225 [02:36<05:54,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 31%|█████████████████████████▏                                                       | 70/225 [02:38<05:43,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 32%|█████████████████████████▌                                                       | 71/225 [02:40<05:49,  2.27s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 32%|█████████████████████████▉                                                       | 72/225 [02:42<05:42,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 32%|██████████████████████████▎                                                      | 73/225 [02:45<05:47,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 33%|██████████████████████████▋                                                      | 74/225 [02:47<06:03,  2.41s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 33%|███████████████████████████                                                      | 75/225 [02:49<05:46,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 34%|███████████████████████████▎                                                     | 76/225 [02:51<05:30,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 34%|███████████████████████████▋                                                     | 77/225 [02:54<05:28,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 35%|████████████████████████████                                                     | 78/225 [02:56<05:27,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 35%|████████████████████████████▍                                                    | 79/225 [02:58<05:23,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 36%|████████████████████████████▊                                                    | 80/225 [03:00<05:23,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 36%|█████████████████████████████▏                                                   | 81/225 [03:03<05:25,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 36%|█████████████████████████████▌                                                   | 82/225 [03:05<05:35,  2.34s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 37%|█████████████████████████████▉                                                   | 83/225 [03:07<05:17,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 37%|██████████████████████████████▏                                                  | 84/225 [03:09<05:13,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 38%|██████████████████████████████▌                                                  | 85/225 [03:12<05:13,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 38%|██████████████████████████████▉                                                  | 86/225 [03:14<05:21,  2.32s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 39%|███████████████████████████████▎                                                 | 87/225 [03:17<05:23,  2.35s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 39%|███████████████████████████████▋                                                 | 88/225 [03:19<05:17,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 40%|████████████████████████████████                                                 | 89/225 [03:21<05:06,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 40%|████████████████████████████████▍                                                | 90/225 [03:23<05:01,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 40%|████████████████████████████████▊                                                | 91/225 [03:26<05:07,  2.29s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 41%|█████████████████████████████████                                                | 92/225 [03:28<05:09,  2.33s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 41%|█████████████████████████████████▍                                               | 93/225 [03:30<04:57,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 42%|█████████████████████████████████▊                                               | 94/225 [03:33<05:06,  2.34s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 42%|██████████████████████████████████▏                                              | 95/225 [03:35<05:02,  2.33s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 43%|██████████████████████████████████▌                                              | 96/225 [03:37<04:51,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 43%|██████████████████████████████████▉                                              | 97/225 [03:39<04:44,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 44%|███████████████████████████████████▎                                             | 98/225 [03:41<04:45,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 44%|███████████████████████████████████▋                                             | 99/225 [03:44<04:49,  2.30s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 44%|███████████████████████████████████▌                                            | 100/225 [03:46<04:39,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 45%|███████████████████████████████████▉                                            | 101/225 [03:48<04:37,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 45%|████████████████████████████████████▎                                           | 102/225 [03:50<04:32,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 46%|████████████████████████████████████▌                                           | 103/225 [03:53<04:33,  2.24s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 46%|████████████████████████████████████▉                                           | 104/225 [03:55<04:25,  2.20s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 47%|█████████████████████████████████████▎                                          | 105/225 [03:57<04:22,  2.19s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 47%|█████████████████████████████████████▋                                          | 106/225 [03:59<04:27,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 48%|██████████████████████████████████████                                          | 107/225 [04:01<04:20,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 48%|██████████████████████████████████████▍                                         | 108/225 [04:04<04:20,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 48%|██████████████████████████████████████▊                                         | 109/225 [04:06<04:17,  2.22s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 49%|███████████████████████████████████████                                         | 110/225 [04:08<04:14,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 49%|███████████████████████████████████████▍                                        | 111/225 [04:10<04:16,  2.25s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 50%|███████████████████████████████████████▊                                        | 112/225 [04:13<04:23,  2.33s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 50%|████████████████████████████████████████▏                                       | 113/225 [04:15<04:18,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 51%|████████████████████████████████████████▌                                       | 114/225 [04:18<04:16,  2.31s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 51%|████████████████████████████████████████▉                                       | 115/225 [04:20<04:08,  2.26s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 52%|█████████████████████████████████████████▏                                      | 116/225 [04:22<04:10,  2.29s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 52%|█████████████████████████████████████████▌                                      | 117/225 [04:24<04:00,  2.23s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 52%|█████████████████████████████████████████▉                                      | 118/225 [04:26<03:56,  2.21s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 53%|██████████████████████████████████████████▎                                     | 119/225 [04:29<04:21,  2.47s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 53%|██████████████████████████████████████████▋                                     | 120/225 [04:32<04:12,  2.40s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 54%|███████████████████████████████████████████                                     | 121/225 [04:34<04:14,  2.45s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 54%|███████████████████████████████████████████▍                                    | 122/225 [04:36<03:58,  2.32s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 55%|███████████████████████████████████████████▋                                    | 123/225 [04:39<03:57,  2.33s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 55%|████████████████████████████████████████████                                    | 124/225 [04:41<04:08,  2.46s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 56%|████████████████████████████████████████████▍                                   | 125/225 [04:44<04:09,  2.49s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 56%|████████████████████████████████████████████▊                                   | 126/225 [04:46<03:46,  2.28s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 56%|█████████████████████████████████████████████▏                                  | 127/225 [04:48<03:30,  2.15s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 57%|█████████████████████████████████████████████▌                                  | 128/225 [04:49<03:15,  2.02s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 57%|█████████████████████████████████████████████▊                                  | 129/225 [04:51<03:14,  2.03s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 58%|██████████████████████████████████████████████▏                                 | 130/225 [04:53<03:08,  1.98s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 58%|██████████████████████████████████████████████▌                                 | 131/225 [04:55<03:11,  2.03s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 59%|██████████████████████████████████████████████▉                                 | 132/225 [04:57<03:08,  2.02s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 59%|███████████████████████████████████████████████▎                                | 133/225 [04:59<02:54,  1.90s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 60%|███████████████████████████████████████████████▋                                | 134/225 [05:01<02:45,  1.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 60%|████████████████████████████████████████████████                                | 135/225 [05:02<02:41,  1.79s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 60%|████████████████████████████████████████████████▎                               | 136/225 [05:04<02:46,  1.87s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 61%|████████████████████████████████████████████████▋                               | 137/225 [05:06<02:39,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 61%|█████████████████████████████████████████████████                               | 138/225 [05:08<02:38,  1.83s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 62%|█████████████████████████████████████████████████▍                              | 139/225 [05:10<02:32,  1.77s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 62%|█████████████████████████████████████████████████▊                              | 140/225 [05:11<02:27,  1.74s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 63%|██████████████████████████████████████████████████▏                             | 141/225 [05:13<02:26,  1.74s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 63%|██████████████████████████████████████████████████▍                             | 142/225 [05:15<02:21,  1.71s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 64%|██████████████████████████████████████████████████▊                             | 143/225 [05:17<02:26,  1.79s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 64%|███████████████████████████████████████████████████▏                            | 144/225 [05:19<02:34,  1.91s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 64%|███████████████████████████████████████████████████▌                            | 145/225 [05:20<02:28,  1.85s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 65%|███████████████████████████████████████████████████▉                            | 146/225 [05:22<02:19,  1.77s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 65%|████████████████████████████████████████████████████▎                           | 147/225 [05:24<02:15,  1.74s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 66%|████████████████████████████████████████████████████▌                           | 148/225 [05:25<02:15,  1.75s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 66%|████████████████████████████████████████████████████▉                           | 149/225 [05:28<02:31,  1.99s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 67%|█████████████████████████████████████████████████████▎                          | 150/225 [05:30<02:27,  1.97s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 67%|█████████████████████████████████████████████████████▋                          | 151/225 [05:32<02:22,  1.93s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 68%|██████████████████████████████████████████████████████                          | 152/225 [05:34<02:16,  1.87s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 68%|██████████████████████████████████████████████████████▍                         | 153/225 [05:35<02:09,  1.80s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 68%|██████████████████████████████████████████████████████▊                         | 154/225 [05:37<02:15,  1.91s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 69%|███████████████████████████████████████████████████████                         | 155/225 [05:39<02:15,  1.93s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 69%|███████████████████████████████████████████████████████▍                        | 156/225 [05:41<02:10,  1.90s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 70%|███████████████████████████████████████████████████████▊                        | 157/225 [05:43<02:08,  1.89s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 70%|████████████████████████████████████████████████████████▏                       | 158/225 [05:45<02:01,  1.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 71%|████████████████████████████████████████████████████████▌                       | 159/225 [05:47<02:08,  1.94s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 71%|████████████████████████████████████████████████████████▉                       | 160/225 [05:49<02:04,  1.91s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 72%|█████████████████████████████████████████████████████████▏                      | 161/225 [05:50<01:56,  1.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 72%|█████████████████████████████████████████████████████████▌                      | 162/225 [05:52<01:55,  1.83s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 72%|█████████████████████████████████████████████████████████▉                      | 163/225 [05:54<01:54,  1.85s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 73%|██████████████████████████████████████████████████████████▎                     | 164/225 [05:56<01:50,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 73%|██████████████████████████████████████████████████████████▋                     | 165/225 [05:58<01:47,  1.79s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 74%|███████████████████████████████████████████████████████████                     | 166/225 [05:59<01:48,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 74%|███████████████████████████████████████████████████████████▍                    | 167/225 [06:01<01:45,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 75%|███████████████████████████████████████████████████████████▋                    | 168/225 [06:03<01:43,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 75%|████████████████████████████████████████████████████████████                    | 169/225 [06:05<01:39,  1.78s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 76%|████████████████████████████████████████████████████████████▍                   | 170/225 [06:06<01:34,  1.72s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 76%|████████████████████████████████████████████████████████████▊                   | 171/225 [06:08<01:35,  1.77s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 76%|█████████████████████████████████████████████████████████████▏                  | 172/225 [06:10<01:34,  1.78s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 77%|█████████████████████████████████████████████████████████████▌                  | 173/225 [06:12<01:33,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 77%|█████████████████████████████████████████████████████████████▊                  | 174/225 [06:14<01:33,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 78%|██████████████████████████████████████████████████████████████▏                 | 175/225 [06:16<01:32,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 78%|██████████████████████████████████████████████████████████████▌                 | 176/225 [06:18<01:32,  1.89s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 79%|██████████████████████████████████████████████████████████████▉                 | 177/225 [06:20<01:35,  2.00s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 79%|███████████████████████████████████████████████████████████████▎                | 178/225 [06:22<01:30,  1.92s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 80%|███████████████████████████████████████████████████████████████▋                | 179/225 [06:23<01:24,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 80%|████████████████████████████████████████████████████████████████                | 180/225 [06:26<01:33,  2.07s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 80%|████████████████████████████████████████████████████████████████▎               | 181/225 [06:34<02:45,  3.77s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 81%|████████████████████████████████████████████████████████████████▋               | 182/225 [06:35<02:14,  3.13s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 81%|█████████████████████████████████████████████████████████████████               | 183/225 [06:37<01:54,  2.73s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 82%|█████████████████████████████████████████████████████████████████▍              | 184/225 [06:39<01:37,  2.37s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 82%|█████████████████████████████████████████████████████████████████▊              | 185/225 [06:40<01:26,  2.16s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 83%|██████████████████████████████████████████████████████████████████▏             | 186/225 [06:42<01:18,  2.03s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 83%|██████████████████████████████████████████████████████████████████▍             | 187/225 [06:43<01:09,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 84%|██████████████████████████████████████████████████████████████████▊             | 188/225 [06:45<01:07,  1.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 84%|███████████████████████████████████████████████████████████████████▏            | 189/225 [06:47<01:05,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 84%|███████████████████████████████████████████████████████████████████▌            | 190/225 [06:49<01:01,  1.77s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 85%|███████████████████████████████████████████████████████████████████▉            | 191/225 [06:50<00:59,  1.74s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 85%|████████████████████████████████████████████████████████████████████▎           | 192/225 [06:52<00:57,  1.74s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 86%|████████████████████████████████████████████████████████████████████▌           | 193/225 [06:54<00:54,  1.71s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 86%|████████████████████████████████████████████████████████████████████▉           | 194/225 [06:55<00:52,  1.68s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 87%|█████████████████████████████████████████████████████████████████████▎          | 195/225 [06:57<00:50,  1.68s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 87%|█████████████████████████████████████████████████████████████████████▋          | 196/225 [06:59<00:49,  1.72s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 88%|██████████████████████████████████████████████████████████████████████          | 197/225 [07:00<00:46,  1.67s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 88%|██████████████████████████████████████████████████████████████████████▍         | 198/225 [07:02<00:45,  1.69s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 88%|██████████████████████████████████████████████████████████████████████▊         | 199/225 [07:04<00:46,  1.79s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 89%|███████████████████████████████████████████████████████████████████████         | 200/225 [07:06<00:46,  1.84s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 89%|███████████████████████████████████████████████████████████████████████▍        | 201/225 [07:08<00:44,  1.83s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 90%|███████████████████████████████████████████████████████████████████████▊        | 202/225 [07:09<00:40,  1.76s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 90%|████████████████████████████████████████████████████████████████████████▏       | 203/225 [07:11<00:38,  1.76s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 91%|████████████████████████████████████████████████████████████████████████▌       | 204/225 [07:13<00:38,  1.82s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 91%|████████████████████████████████████████████████████████████████████████▉       | 205/225 [07:15<00:38,  1.95s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 92%|█████████████████████████████████████████████████████████████████████████▏      | 206/225 [07:17<00:37,  1.99s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 92%|█████████████████████████████████████████████████████████████████████████▌      | 207/225 [07:19<00:35,  1.98s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 92%|█████████████████████████████████████████████████████████████████████████▉      | 208/225 [07:21<00:33,  1.99s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 93%|██████████████████████████████████████████████████████████████████████████▎     | 209/225 [07:23<00:32,  2.00s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 93%|██████████████████████████████████████████████████████████████████████████▋     | 210/225 [07:27<00:36,  2.47s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 94%|███████████████████████████████████████████████████████████████████████████     | 211/225 [07:33<00:47,  3.41s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 94%|███████████████████████████████████████████████████████████████████████████▍    | 212/225 [07:34<00:38,  2.93s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 95%|███████████████████████████████████████████████████████████████████████████▋    | 213/225 [07:36<00:30,  2.57s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 95%|████████████████████████████████████████████████████████████████████████████    | 214/225 [07:38<00:25,  2.29s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 96%|████████████████████████████████████████████████████████████████████████████▍   | 215/225 [07:40<00:21,  2.16s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 96%|████████████████████████████████████████████████████████████████████████████▊   | 216/225 [07:41<00:17,  1.98s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 96%|█████████████████████████████████████████████████████████████████████████████▏  | 217/225 [07:43<00:15,  1.92s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 97%|█████████████████████████████████████████████████████████████████████████████▌  | 218/225 [07:45<00:13,  1.89s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 97%|█████████████████████████████████████████████████████████████████████████████▊  | 219/225 [07:47<00:11,  1.88s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 98%|██████████████████████████████████████████████████████████████████████████████▏ | 220/225 [07:48<00:09,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 98%|██████████████████████████████████████████████████████████████████████████████▌ | 221/225 [07:50<00:07,  1.91s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 99%|██████████████████████████████████████████████████████████████████████████████▉ | 222/225 [07:52<00:05,  1.88s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


 99%|███████████████████████████████████████████████████████████████████████████████▎| 223/225 [07:54<00:03,  1.85s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


100%|███████████████████████████████████████████████████████████████████████████████▋| 224/225 [07:56<00:01,  1.81s/it]

INFO:tensorflow:Restoring parameters from vggish_model.ckpt


100%|████████████████████████████████████████████████████████████████████████████████| 225/225 [07:58<00:00,  2.13s/it]


In [17]:
# computing mean and standard deviation throughout the entire segment
vggish_stat= vggish_processing.vggish_stat(vggish)
vggish_mean = vggish_stat.mean()
vggish_std = vggish_stat.std()

C:\Users\deepd\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3334: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
C:\Users\deepd\Anaconda3\lib\site-packages\numpy\core\_methods.py:161: RuntimeWarning: invalid value encountered in true_divide
  ret = ret.dtype.type(ret / rcount)
C:\Users\deepd\Anaconda3\lib\site-packages\numpy\core\_methods.py:216: RuntimeWarning: Degrees of freedom <= 0 for slice
  ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
C:\Users\deepd\Anaconda3\lib\site-packages\numpy\core\_methods.py:185: RuntimeWarning: invalid value encountered in true_divide
  arrmean = um.true_divide(
C:\Users\deepd\Anaconda3\lib\site-packages\numpy\core\_methods.py:209: RuntimeWarning: invalid value encountered in true_divide
  ret = ret.dtype.type(ret / rcount)


In [18]:
# combining those feature to a dataframe
df_vggish_mean = pd.DataFrame(vggish_mean,columns=['vggish_feature_mean']*128)
df_vggish_std = pd.DataFrame(vggish_mean,columns=['vggish_feature_std']*128)
vggish_features = df_vggish_mean.join(df_vggish_std)

In [19]:
vggish_features.head().T

0         1         2         3         4
vggish_feature_mean -0.014604  0.186208 -0.089034 -0.112447 -0.014857
vggish_feature_mean -0.195043 -0.357018 -0.150178 -0.258087 -0.169344
vggish_feature_mean  0.123979  0.180552  0.552570  1.109780  0.456923
vggish_feature_mean -0.350942 -0.404429 -0.190024 -0.567602 -0.106327
vggish_feature_mean -0.392104 -0.086799 -0.092157 -0.118912 -0.021436
...                       ...       ...       ...       ...       ...
vggish_feature_std   0.117038  0.428696  0.239576  0.041459  0.409458
vggish_feature_std  -0.126274  0.056640 -0.025697 -0.169521 -0.040775
vggish_feature_std  -0.092976 -0.087464 -0.154079 -0.388355 -0.164079
vggish_feature_std  -0.212546 -0.209597 -0.430225 -0.139727 -0.216557
vggish_feature_std  -0.543742 -0.335151 -0.288353 -0.321753 -0.249364

[256 rows x 5 columns]

In [20]:
vggish_features.shape

(225, 256)

In [21]:
vggish_features.to_csv('vggish_features.csv',index=True)

In [22]:
# joining vggish features and MFCC features into one dataframe
vggish_with_mfcc = mfcc_feature.join(vggish_features)

In [23]:
vggish_with_mfcc.head().T

0           1           2           3           4
mfcc_mean_feature  -765.857361 -673.794434 -718.185974 -771.010437 -652.407043
mfcc_mean_feature    56.635921   39.070114   66.435989   65.506836   99.792557
mfcc_mean_feature    -7.718951  -55.959335   -7.685933   10.993238  -42.095676
mfcc_mean_feature    -5.349398   -3.420140   21.239208   14.352894   30.789314
mfcc_mean_feature    -2.479132  -32.085152  -40.569927  -20.039225  -26.789732
...                        ...         ...         ...         ...         ...
vggish_feature_std    0.117038    0.428696    0.239576    0.041459    0.409458
vggish_feature_std   -0.126274    0.056640   -0.025697   -0.169521   -0.040775
vggish_feature_std   -0.092976   -0.087464   -0.154079   -0.388355   -0.164079
vggish_feature_std   -0.212546   -0.209597   -0.430225   -0.139727   -0.216557
vggish_feature_std   -0.543742   -0.335151   -0.288353   -0.321753   -0.249364

[685 rows x 5 columns]

In [24]:
vggish_with_mfcc.shape

(225, 685)

## Extracting spectral features and statistics

In [25]:
# extracting frame-level features
cough_heavy['spectral_centroid'] = preprocessing.Extract_Data.get_spectral_centroid(cough_heavy)
cough_heavy['roll_off'] = preprocessing.Extract_Data.roll_off(cough_heavy)
cough_heavy['zero_crossing_rate'] = preprocessing.Extract_Data.get_zero_crossing(cough_heavy)
cough_heavy['rms'] = preprocessing.Extract_Data.get_zero_crossing(cough_heavy)

In [26]:
# computing statistics from the frame-level features
preprocessing.postprocess(cough_heavy,labels=['spectral_centroid','roll_off','zero_crossing_rate','rms'])

In [59]:
# having onset as feature
cough_heavy['onset'] = preprocessing.Extract_Data.onset(cough_heavy)

In [75]:
cough_heavy.head()

label                                               data  \
0   0.00  [1.141661e-05, 3.862317e-06, -3.9919647e-05, -...   
1   1.00  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
2   1.00  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
3   0.00  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   
4   0.00  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...   

                                            filename  \
0  C:/Users/deepd/Documents/Covid/Extracted_trial...   
1  C:/Users/deepd/Documents/Covid/Extracted_trial...   
2  C:/Users/deepd/Documents/Covid/Extracted_trial...   
3  C:/Users/deepd/Documents/Covid/Extracted_trial...   
4  C:/Users/deepd/Documents/Covid/Extracted_trial...   

                                   spectral_centroid  \
0  [[1661.6969269932445, 1597.5467369335083, 1626...   
1  [[0.0, 0.0, 0.0, 0.0, 0.0, 3620.666656334686, ...   
2  [[5339.464481751565, 3698.983581618055, 3744.7...   
3  [[1900.433774135117, 1696.7200733662612, 1685....   
4  [[0.0, 0.0, 3248.724186007833, 2277.0602467323...   

                                            roll_off  \
0  [[2950.048828125, 2670.1171875, 2767.016601562...   
1  [[0.0, 0.0, 0.0, 0.0, 0.0, 6556.8603515625, 63...   
2  [[8936.279296875, 6373.828125, 6621.4599609375...   
3  [[4446.6064453125, 3811.376953125, 3919.042968...   
4  [[0.0, 0.0, 5447.900390625, 4640.4052734375, 4...   

                                  zero_crossing_rate  \
0  [[0.041015625, 0.07080078125, 0.0869140625, 0....   
1  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.11181640625, 0.20...   
2  [[0.025390625, 0.185546875, 0.3984375, 0.57226...   
3  [[0.0263671875, 0.056640625, 0.056640625, 0.05...   
4  [[0.0, 0.0, 0.02734375, 0.0712890625, 0.112792...   

                                                 rms  spectral_centroid_mean  \
0  [[0.041015625, 0.07080078125, 0.0869140625, 0....                 2067.38   
1  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.11181640625, 0.20...                 2455.76   
2  [[0.025390625, 0.185546875, 0.3984375, 0.57226...                 2934.96   
3  [[0.0263671875, 0.056640625, 0.056640625, 0.05...                 2288.21   
4  [[0.0, 0.0, 0.02734375, 0.0712890625, 0.112792...                 1901.79   

   spectral_centroid_std  spectral_centroid_median  ...  rms_median  rms_max  \
0                 595.30                   2076.12  ...        0.15     0.50   
1                1035.04                   2692.79  ...        0.19     0.45   
2                1107.07                   3118.39  ...        0.30     0.86   
3                 967.49                   2138.40  ...        0.14     0.39   
4                 601.62                   1811.02  ...        0.12     0.39   

   rms_min  rms_quartile_1st  rms_quartile_3rd  rms_interquartilerange  \
0     0.03              0.09              0.28                    0.18   
1     0.00              0.12              0.24                    0.12   
2     0.03              0.20              0.51                    0.31   
3     0.03              0.10              0.27                    0.16   
4     0.00              0.09              0.17                    0.09   

   rms_skewness  rms_kurtosis  rms_rms  onset  
0          0.93         -0.49     0.24     11  
1         -0.01         -0.30     0.20     14  
2          0.66         -0.53     0.42     11  
3          0.45         -1.13     0.21      8  
4          1.05          0.40     0.16     13  

[5 rows x 52 columns]

In [61]:
cough_heavy.shape

(225, 52)

In [62]:
cough_data = cough_heavy.copy()

In [63]:
cough_data.drop(['spectral_centroid','roll_off','zero_crossing_rate','rms','data','filename'],axis=1,inplace=True)

In [64]:
cough_data.shape

(225, 46)

In [65]:
# joining all features
cough_data = cough_data.join(vggish_with_mfcc)

In [66]:
cough_data.shape

(225, 731)

In [67]:
#dropping missing value
cough_data.dropna(inplace=True)

In [68]:
cough_data.to_csv('all_features.csv',index=False)

In [70]:
cough_data.columns = range(0,731)

In [71]:
cough_data.head()

0       1       2       3       4      5       6       7       8     9    \
0 0.00 2067.38  595.30 2076.12 4179.15 708.67 1629.15 2470.99  841.84  0.49   
1 1.00 2455.76 1035.04 2692.79 4629.34   0.00 1866.12 3267.20 1401.09 -0.76   
2 1.00 2934.96 1107.07 3118.39 5339.46 743.43 2247.07 3711.15 1464.08 -0.22   
3 0.00 2288.21  967.49 2138.40 3956.37 467.76 1524.06 3197.04 1672.97  0.01   
4 0.00 1901.79  601.62 1811.02 3604.11   0.00 1466.70 2274.14  807.44  0.46   

   ...   721   722  723   724   725  726   727   728   729   730  
0  ... -0.78 -0.17 0.09 -0.37 -0.25 0.12 -0.13 -0.09 -0.21 -0.54  
1  ... -0.46  0.21 0.16 -0.31 -0.25 0.43  0.06 -0.09 -0.21 -0.34  
2  ... -0.68 -0.10 0.19 -0.32 -0.07 0.24 -0.03 -0.15 -0.43 -0.29  
3  ... -0.65 -0.14 0.08 -0.35 -0.04 0.04 -0.17 -0.39 -0.14 -0.32  
4  ... -0.52  0.13 0.03 -0.08 -0.16 0.41 -0.04 -0.16 -0.22 -0.25  

[5 rows x 731 columns]

### Fitting a basic model with all the features

In [73]:
# splitting the rows for modelling
from sklearn.model_selection import train_test_split
features = cough_data.drop([0],axis=1)
labels = cough_data[0]
train_features,validation_features,train_label,validation_label = train_test_split(features,labels,test_size=.2,random_state=123)

In [136]:
from sklearn.ensemble import RandomForestClassifier
rf = RandomForestClassifier(n_jobs=-1,random_state=3,n_estimators=210)
rf.fit(train_features,train_label)

RandomForestClassifier(n_estimators=210, n_jobs=-1, random_state=3)

In [137]:
pred = rf.predict(validation_features)

In [138]:
from sklearn.metrics import accuracy_score
accuracy = accuracy_score(validation_label,pred)

In [139]:
accuracy

0.7692307692307693

### Conclusion

So, this basic model gives a accuracy of approximately 77%, which definately shows promise. Also, I did not distingush cold cough, asthama cough from covid for generalization, doing this will also give significant amoount of accuracy. Appart from that we didnot have enough data, so we could not proceed with any deep learning technique. I will try to increse this accuracy with fitting differents type models and aslo with Data Augmentation.